src/optimization/optimizer.ml

```diff
@@ -672,6 +672,32 @@ let optimize_completion_expr e args =
 			let efor = loop efor in
 			old();
 			(EFor ((EBinop (OpIn,id,it),p),efor),p)
+		| EFor ((EBinop (OpArrow,((EConst (Ident key),_) as kid),(EBinop (OpIn,((EConst (Ident value),_) as vid),it),_)),p),efor) ->
```

Comment:
That's a very specific pattern. Shouldn't we just check if the display position is within the `EFor` and leave its "header" alone in that case? We can still "optimize" the loop body normally.

Reply:
But we need to extract `key` and `value` identifiers to declare them. Hence the pattern.
The same approach is already used for "classic" `for(value in collection)`: 
https://github.com/HaxeFoundation/haxe/blob/1d45ab0419442f9327ee0efe177dc3a4675c5622/src/optimization/optimizer.ml#L663-L666

Reply:
I'm not sure it's a good idea there either, but at least the pattern there is quite simple and unlikely to change. Key-value iteration relies on that funky parsing precedence and if we ever fix that the detection here would break.

I'm fine with merging this but I think we should look into making this thing more robust in general. IMO just checking if the display position is in `EFor(e1,e2)` and then detecting all `Ident`s in `e1` would be better. I'm sure there can be cases where that leads to some false positives, but those hardly matter in this "optimization".

Anyway, feel free to merge for now.

Reply:
I've changed it to detect all idents. Check again, please.